### PR TITLE
New Activity page

### PR DESCRIPTION
**Pull request by ace0625**
- Mainly update activity page with two tabs (new requests/connected).

- new request page will be optimized momentarily with flexible rows.

- Fix typos.

### CODE REVIEW

File: src/components/Activity/ConnectedRow.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  TouchableWithoutFeedback,
+  View,
+} from 'react-native';
+import { Actions } from 'react-native-router-flux';
+import ChatPage from '../Chat/ChatPage';
+
+class ConnectedRow extends Component {
+  constructor(props) {
+    super(props);
+    console.log(this.props.dataSource);
```
Comment (lyw0149) [inline src/components/Activity/ConnectedRow.js]: console.log

File: src/components/Activity/NewRequests.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      isEmpty: newRequests.length === 0,
+    });
+  }
+
+  renderRow(rowData, sectionID, rowID) {
+    return <NewRequestsRow dataSource={rowData} onSelect={this.onRequestSuccess.bind(this)} id={rowID}/>;
+  }
+
+  renderSeparator(sectionID, rowID) {
+    console.log(rowID);
```
Comment (lyw0149) [inline src/components/Activity/NewRequests.js]: console.log

File: src/components/Activity/NewRequestsRow.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      height: 70,
+    };
+
+    ServerUtil.initCallback(
+      (result) => this.onRequestSuccess(result),
+      (error) => this.onRequestFail(error));
+  }
+
+  componentWillMount() {
+    console.log(this.props.id);
```
Comment (lyw0149) [inline src/components/Activity/NewRequestsRow.js]: console.log

File: src/components/Activity/NewRequestsRow.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  render() {
+    const SwipeoutButtons = [
+      {
+        text: 'Delete',
+        backgroundColor: '#fd5b52',
+        onPress: this.rejectRequest.bind(this),
+      },
+    ];
+
+    console.log(this.state.height);
```
Comment (lyw0149) [inline src/components/Activity/NewRequestsRow.js]: console.log

File: src/components/Activity/Swipeout.js [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+      </TouchableHighlight>
+    )
+  }
+})
+
+var Swipeout = React.createClass({
+  mixins: [tweenState.Mixin]
+  , getDefaultProps: function() {
+    return {
+      onOpen: function(sectionID, rowID) {console.log('onOpen: '+sectionID+" "+rowID)},
```
Comment (lyw0149) [inline src/components/Activity/Swipeout.js:74]: console.log

File: src/components/Activity/ConnectedRow.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+      <ActivityIndicator
+        animating={!this.state.loaded}
+        style={[styles.activityIndicator]}
+        size='small'
+      />
+    );
+  }
+
+  onChatButtonPress() {
+    // Alert.alert('onChatPress');
```
Comment (lyw0149) [inline src/components/Activity/ConnectedRow.js]: alert comment

**Review by lyw0149 — COMMENTED** [review]
Few comments below. Thank you @ace0625 @ifndefdeadmau5.

**Comment by lyw0149** [comment]
LGTM Thanks. :shipit:

File: src/components/Activity/Activity.js [hunk 7]
```diff
@@ -8,176 +8,48 @@ import {
...
-      (error) => this.onRequestFail(error));
-
     this.state = {
-        dataBlob: {},
-        dataSource: new ListView.DataSource({
-          rowHasChanged: (r1, r2) => r1 !== r2,
-          sectionHeaderHasChanged: (s1, s2) => s1 !== s2,
-        }),
         loaded: false,
         isRefreshing: false,
```
Comment (ifndefdeadmau5) [inline src/components/Activity/Activity.js:23]: Are we going to use loaded, isRefreshing state? or we can just remove them

**Comment by lyw0149** [comment]
LGTM

**Pull request merged**